src/hotspot/share/gc/shared/barrierSetNMethod.cpp

```diff
@@ -67,10 +61,26 @@ bool BarrierSetNMethod::supports_entry_barrier(nmethod* nm) {
 }
 
 bool BarrierSetNMethod::nmethod_entry_barrier(nmethod* nm) {
+  class OopKeepAliveClosure : public OopClosure {
+  public:
+    OopKeepAliveClosure() {}
```

Comment:
The constructor could be removed as well now. Still looks good.

Reply:
Fixed. Thanks Thomas.